src/input.rs

```diff
@@ -63,10 +67,10 @@ impl InputHandler {
                 }
                 _ => {
                     self.send_screen_instructions
-                        .send(ScreenInstruction::ClearScroll)
+                        .send((ScreenInstruction::ClearScroll, err_ctx))
```

Comment:
So, this is true to other places as well, but what do you think about making a wrapper struct around our various `Sender`s so that we won't have to keep specifying the `err_ctx` everywhere?

Something like:
```rust
struct SenderWithContext<T> {
    err_ctx: ErrorContext,
    sender: Sender<T>
}
impl<T> SenderWithContext <T>{
    pub fn send(event: T) {
        self.sender.send(event, self.err_ctx);
    }
    pub fn update_error_ctx (new_error_ctx: ErrorContext) {
        self.error_ctx = new_error_ctx;
    }
}
```

Reply:
This is a good idea. What about the `SyncSender`?

Reply:
Hmm, good question. Either have a `SyncSenderWithContext` version or maybe have `self.sender` be an `enum` somehow? Whatever you feel will look/work better. I think this is one of those cases where developer ergonomics is important.

Reply:
I have updated this. The only trivial drawback here was that the sender in panic hook cannot be updated because that requires mutable borrow of the sender, but the closure cannot capture the sender mutably. But I don't think it matters much because that info will be displayed only if another panic occurred following an invocation of panic hook, which is rare. The info of the primary panic is still displayed correctly.